samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -132,7 +132,7 @@ public void testStreamProcessorWithNoTask() {
     configMap.remove("task.class");
     final Config configs = new MapConfig(configMap);
 
-    StreamProcessor processor = new StreamProcessor(1, configs, new HashMap<>());
```

Comment:
Ok. Let me take another look at adding tests. 

>  Again not necessarily related to this specific commit, but I would also test start / start, start / stop / stop, and stop without start.

Initial implementation goals were to allow start and stop on a processor only once. Hence, I remember adding the tests in the LinkedIn patch. But I guess code has evolved quite a bit and I missed those tests. I will address that case as well.